File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.cpp
#include <bits/stdc++.h>
using namespace std;
int solution(vector<int> scoville, int k) {
    priority_queue<int, vector<int>, greater<int
    > > pq;
    for (int s: scoville){
        pq.push(s);
    }
    int cnt = 0;
    while (pq.size() >= 2 && pq.top() < k){
        int a = pq.top();
        pq.pop();
        int b = pq.top();
        pq.pop();
        pq.push(a + b * 2);
        cnt++;        
    }
    if (pq.top() < k)
        return -1;
    return cnt;
}

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq


def solution(scoville, k):
    hq = []
    cnt = 0
    for s in scoville:
        heapq.heappush(hq, s)
    while len(hq) >= 2 and hq[0] < k:
        cnt += 1
        heapq.heappush(hq, heapq.heappop(hq) + heapq.heappop(hq) * 2)
    if hq[0] < k:
        return -1
    else: 
        return cnt
